### ISSUE TITLE
adding tag\name for a rule

### ISSUE THREAD

**Issue by avishni01** [terraform-aws-modules/terraform-aws-security-group#327]
hello

i have just start using the module instead of creating with the basic terraform resources. it took me time to use it correctly , but one thing is missing (or maybe i didn't find how to do it), naming the SG rules. the name is actully bases on the tag Name, vurrntly in the default view all rules are displayed with empty name , there should be an option to add tags and name to for each rule.

**Comment by FlorinAndrei** [comment]
Yeah, there might be issues when names are conflicting with each other, but this would be very useful. I actually found this ticket because I was looking for a way to tag or describe the rules.

**Comment by ScubaDrew** [comment]
This module is not that great. Adding/Changing rules never works correctly either.

**Comment by robertalexa** [comment]
According to documentation, it think it is just a case of using a different resource for rules
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule

Please note the first note:
![image](https://github.com/user-attachments/assets/da700dc4-1cda-4572-8bf4-6b0826790977)

I interpret that as the module (this as well as others e.g. ecs) to drop the use of `aws_security_group_rule` and switch to `aws_vpc_security_group_egress_rule` and `aws_vpc_security_group_ingress_rule` respectively.

Wondering if maybe @antonbabenko or @bryantbiggs know anything on this or are tracking this on their roadmap already.

**Comment by tomasz-baran** [comment]
No stale.

**Comment by tigpt** [comment]
We need to upgrade to `aws_vpc_security_group_ingress_rule` and to `aws_vpc_security_group_ingress_rule` in order to use tags on the SG rules. If I want to detect drifts in sg rules I normally use AWS Tag Editor to export all resources tags that miss "terraform=true" but the SG rules can't implement tags with the current resource `aws_security_group_rule`.

As [documentation](https://registry.terraform.io/providers/hashicorp/aws/5.82.2/docs/resources/security_group_rule) an upgrade from `aws_security_group_rule` to `aws_vpc_security_group_*gress_rule` is recommended.

**Comment by elstak** [comment]
not stale 👍

**Comment by kirkchong** [comment]
ill pick this up

**Comment by ScubaDrew** [comment]
not stale

**Comment by to-bar** [comment]
not stale

**Comment by kirkchong** [comment]
40% done with the ingress rules, left the egress rules, been busy with work, I'll try to raise the PR once done and get it reviewed, expect a few comments on the PR as its a breaking change

**Comment by antonbabenko** [comment]
This issue has been resolved in version 6.0.0 :tada:
